modules/javafx.graphics/src/main/java/com/sun/javafx/font/PrismFontFactory.java

```diff
@@ -238,6 +238,13 @@ private PrismFontFile createFontResource(String name,
                                              boolean register, boolean embedded,
                                              boolean copy, boolean tracked) {
         String key = (filename+index).toLowerCase();
+        /*
+         * macOS: we need to load unique fonts for regular and bold.
+         * Probably this should be handled elsewhere
```

Comment:
This indeed adds macos specific behavior in an otherwise OS-neutral concept, but I don't see a simple approach where the key is calculated in an OS-specific approach without introducing artificial overhead. So I am ok with doing this here.